### PR TITLE
Added license scanning info

### PR DESCRIPTION
**Pull request by hayleycd**
[Preview here](https://hayleycd.github.io/osv-scanner/experimental/)

### CODE REVIEW

**Comment by josieang** [comment]
I am making a few minor changes to the interface after a discussion with rex https://github.com/google/osv-scanner/issues/671#issuecomment-1823770821 and will review this once that is merged!

**Comment by hayleycd** [comment]
Got it. Thanks @josieang. I will follow up with a few questions tomorrow.

File: docs/experimental.md [hunk 1]
```diff
@@ -128,3 +128,38 @@ Set the location of your manually downloaded database by following the instructi
...
+## License scanning
+
+OSV-Scanner supports license checking as an experimental feature. The data comes from the [deps.dev API](https://docs.deps.dev/api/v3alpha/).
+
+### License summary
+
+If you want a summary of your dependencies licenses, run the following command:
+
+```bash
+osv-scanner --experimental-license path/to/repository`
```
Comment (hayleycd) [inline docs/experimental.md:141]:
```suggestion
osv-scanner --experimental-license-summary path/to/repository`
```

File: docs/experimental.md [hunk 2]
```diff
@@ -128,3 +128,38 @@ Set the location of your manually downloaded database by following the instructi
...
 
 1. Commit level scanning is not supported. 
+
+## License scanning
+
+OSV-Scanner supports license checking as an experimental feature. The data comes from the [deps.dev API](https://docs.deps.dev/api/v3alpha/).
+
+### License summary
+
+If you want a summary of your dependencies licenses, run the following command:
```
Comment (hayleycd) [inline docs/experimental.md:138]:
```suggestion
If you want a summary of your dependencies licenses, use the `--experimental-license-summary` flag:
```

File: docs/experimental.md [hunk 3]
```diff
@@ -128,3 +128,38 @@ Set the location of your manually downloaded database by following the instructi
...
+
+```bash
+osv-scanner --experimental-license path/to/repository`
+```
+
+### License violations
+
+To set an allowed license list and see the details of packages that do not conform, run the following:
```
Comment (hayleycd) [inline docs/experimental.md:146]:
```suggestion
To set an allowed license list and see the details of packages that do not conform, use the `--experimental-license` flag:
```

File: docs/experimental.md [hunk 4]
```diff
@@ -128,3 +128,38 @@ Set the location of your manually downloaded database by following the instructi
...
+### License violations
+
+To set an allowed license list and see the details of packages that do not conform, use the `--experimental-license` flag:
+
+```bash
+osv-scanner --experimental-license="comma separated list of allowed licenses" path/to/directory
+```
+Include your allowed licenses as a comma separated list. OSV-Scanner recognizes licenses in SPDX format. Please indicate your allowed licenses using [SPDX license](https://spdx.org/licenses/) identifiers. 
+
+### License violations example
```
Comment (hayleycd) [inline docs/experimental.md:153]:
```suggestion
#### License violations example
```

File: docs/experimental.md [hunk 5]
```diff
@@ -128,3 +128,38 @@ Set the location of your manually downloaded database by following the instructi
...
+If you wanted to allow the following licenses:
+
+- [BSD 3-Clause "New" or "Revised" License](https://spdx.org/licenses/BSD-3-Clause.html)
+- [Apache License 2.0](https://spdx.org/licenses/Apache-2.0.html)
+- [MIT](https://spdx.org/licenses/MIT.html)
+
+Your command would be in this form:
+
+```bash
+osv-scanner --experimental-license="BSD-3-Clause,Apache-2.0, MIT" path/to/directory
```
Comment (hayleycd) [inline docs/experimental.md:164]:
```suggestion
osv-scanner --experimental-license="BSD-3-Clause,Apache-2.0,MIT" path/to/directory
```

File: docs/experimental.md [hunk 6]
```diff
@@ -128,3 +128,38 @@ Set the location of your manually downloaded database by following the instructi
 ### Limitations
 
 1. Commit level scanning is not supported. 
+
+## License scanning
+
+OSV-Scanner supports license checking as an experimental feature. The data comes from the [deps.dev API](https://docs.deps.dev/api/v3alpha/).
```
Comment (josieang) [inline docs/experimental.md:134]: I'm not sure what's better, https://docs.deps.dev/api/ or https://docs.deps.dev/api/v3alpha/. I know that /api redirects to the latest version of the api docs (although in this case the only version is v3alpha). We do use v3alpha in this feature at the moment, so I think it's fine.

File: docs/experimental.md [hunk 7]
```diff
@@ -128,3 +128,38 @@ Set the location of your manually downloaded database by following the instructi
...
+```bash
+osv-scanner --experimental-license-summary path/to/repository`
+```
+
+### License violations
+
+To set an allowed license list and see the details of packages that do not conform, use the `--experimental-license` flag:
+
+```bash
+osv-scanner --experimental-license="comma separated list of allowed licenses" path/to/directory
```
Comment (josieang) [inline docs/experimental.md:149]: comma-separated ?

File: docs/experimental.md [hunk 8]
```diff
@@ -128,3 +128,38 @@ Set the location of your manually downloaded database by following the instructi
...
+```
+Include your allowed licenses as a comma separated list. OSV-Scanner recognizes licenses in SPDX format. Please indicate your allowed licenses using [SPDX license](https://spdx.org/licenses/) identifiers.
```
Comment (josieang) [inline docs/experimental.md:151]: ditto

File: docs/experimental.md [hunk 9]
```diff
@@ -128,3 +128,38 @@ Set the location of your manually downloaded database by following the instructi
...
+## License scanning
+
+OSV-Scanner supports license checking as an experimental feature. The data comes from the [deps.dev API](https://docs.deps.dev/api/v3alpha/).
+
+### License summary
+
+If you want a summary of your dependencies licenses, use the `--experimental-license-summary` flag:
+
+```bash
+osv-scanner --experimental-license-summary path/to/repository`
```
Comment (josieang) [inline docs/experimental.md:141]: the flag is pluralised "licenses" everywhere ("experimental-licenses-summary", "experimental-licenses"). do you think this is weird? should be singular "license"? I can't tell anymore from having worked on this feature for so long.

File: docs/experimental.md (hunk 6 above)
Comment (hayleycd) [reply docs/experimental.md:134]: Let's do https://docs.deps.dev/api/. We are not linking to the specific part of the API we are using, but just pointing people to a resource where they can learn more if they would like.

File: docs/experimental.md (hunk 7 above)
Comment (hayleycd) [reply docs/experimental.md:149]: Good catch.

File: docs/experimental.md (hunk 9 above)
Comment (hayleycd) [reply docs/experimental.md:141]: Weirdly I think `--experimental-licenses` makes sense and `--experimental-license-summary` makes sense but we should definitely not have one plural and the other singular. Let's stick with plural. I will fix it.

File: docs/experimental.md (hunk 6 above)
Comment (hayleycd) [inline docs/experimental.md:134]:
```suggestion
OSV-Scanner supports license checking as an experimental feature. The data comes from the [deps.dev API](https://docs.deps.dev/api/).
```

File: docs/experimental.md (hunk 9 above)
Comment (hayleycd) [inline docs/experimental.md:141]:
```suggestion
osv-scanner --experimental-licenses-summary path/to/repository`
```

File: docs/experimental.md [hunk 10]
```diff
@@ -128,3 +128,38 @@ Set the location of your manually downloaded database by following the instructi
...
 
 1. Commit level scanning is not supported. 
+
+## License scanning
+
+OSV-Scanner supports license checking as an experimental feature. The data comes from the [deps.dev API](https://docs.deps.dev/api/v3alpha/).
+
+### License summary
+
+If you want a summary of your dependencies licenses, use the `--experimental-license-summary` flag:
```
Comment (hayleycd) [inline docs/experimental.md:138]:
```suggestion
If you want a summary of your dependencies licenses, use the `--experimental-licenses-summary` flag:
```

File: docs/experimental.md [hunk 11]
```diff
@@ -128,3 +128,38 @@ Set the location of your manually downloaded database by following the instructi
...
+
+```bash
+osv-scanner --experimental-license-summary path/to/repository`
+```
+
+### License violations
+
+To set an allowed license list and see the details of packages that do not conform, use the `--experimental-license` flag:
```
Comment (hayleycd) [inline docs/experimental.md:146]:
```suggestion
To set an allowed license list and see the details of packages that do not conform, use the `--experimental-licenses` flag:
```

File: docs/experimental.md (hunk 7 above)
Comment (hayleycd) [inline docs/experimental.md:149]:
```suggestion
osv-scanner --experimental-license="comma-separated list of allowed licenses" path/to/directory
```

File: docs/experimental.md (hunk 8 above)
Comment (hayleycd) [inline docs/experimental.md:151]:
```suggestion
Include your allowed licenses as a comma-separated list. OSV-Scanner recognizes licenses in SPDX format. Please indicate your allowed licenses using [SPDX license](https://spdx.org/licenses/) identifiers. 
```

**Review by josieang — APPROVED** [review]
thanks for this Hayley!

**Review by oliverchang — APPROVED** [review]

**Pull request merged**